### ISSUE TITLE
Home bridge Stops Working Every Few Hours, Has To Be Reset To Recover, Errors Included

### ISSUE THREAD

**Issue by Wildcat1** [pdlove/homebridge-smartthings#52]
Posting over here too as realized this was probably the best place for this issue,

@pdlove / All
Seeing a new issue pop up and was wondering if others have seen something similar or had any ideas on how to fix:

error at req: { [Error: connect EHOSTUNREACH 107.20.201.51:443 - Local (removed by me from this thread)]
code: 'EHOSTUNREACH',
errno: 'EHOSTUNREACH',
syscall: 'connect',
address: '107.20.201.51',
port: 443 }

then I also see this

error at req: { [Error: getaddrinfo ENOTFOUND graph.api.smartthings.com graph.api.smartthings.com:443]
code: 'ENOTFOUND',
errno: 'ENOTFOUND',
syscall: 'getaddrinfo',
hostname: 'graph.api.smartthings.com',
host: 'graph.api.smartthings.com',
port: '443' }

I see this happen after x number of hours of running and at that time is when I also lose access to the SmartThings devices in HomeKit (running through Homebridge). Restart Homebridge and everything works again but wondering what is causing the issues above.

It certainly is strange because Homebridge won't last even 24 hours before I start seeing those issues and although I still am able to access my Nest through it (through the KraigM port) everything else disappears that was brought in via the JSON complete API. It never recovers from the errors I posted until I stop the script and start fresh, then it runs fine again for a few hours before hitting the same problem. 

Thanks as always for the help

For reference running on OS X 10.11.4 beta 5 and testing against iOS 9.3 beta 5

WC

**Comment by edasque** [comment]
I do see that as well, on a RPi.

**Comment by rickybloomfield** [comment]
Same here on RPi2.

**Comment by pdlove** [comment]
I want to  let everyone know, I'm not ignoring and have not abandoned this project.  I have time set aside in my schedule later tonight and tomorrow to revamp the way in the error trapping works to provide better feedback and make it not  cause a crash or problems with values of accessories

**Comment by A5HRAJ** [comment]
Same issue on my RPi 2. Every time this happens, the only way to fix it is to remove Homebridge from HomeKit, change the username in config.json, and set up my accessories all over again.

**Comment by rickybloomfield** [comment]
Same here.

**Comment by realdannys** [comment]
Same here - log as follows. These errors don't even appear as errors to the running process either - it just breaks Homebridge.

```error at req:  { [Error: getaddrinfo ENOTFOUND graph-eu01-euwest1.api.smartthings.com graph-eu01-euwest1.api.smartthings.com:443]
      code: 'ENOTFOUND',
  errno: 'ENOTFOUND',
  syscall: 'getaddrinfo',
  hostname: 'graph-eu01-euwest1.api.smartthings.com',
  host: 'graph-eu01-euwest1.api.smartthings.com',
  port: '443' }```

**Comment by brandwe** [comment]
Same here as well. Ready to help any way I can!

**Comment by Wildcat1** [comment]
@pdlove thanks for your continued work on this, we all do appreciate it. Let us know if there is anything we can test or info we can provide to help with this effort.

**Comment by realdannys** [comment]
Unofortuantely this bug seems to break Homebridge any where between 10 mins and 60 mins after starting. Tempted to run a script for the moment which restarts homebridge every 5 minutes.

**Comment by thoward1234** [comment]
I have been seeing this as well; additionally 1/4 to 1/2 of my devices vanish.  They re-appear on restart, but their assignment to rooms and scenes are lost (from a Homebridge standpoint; everything is normal for Smartthings)

**Comment by brandwe** [comment]
I've been thinking this could be an API error on SmartThings side? I've noticed errors accessing the actual app as well after this - I wonder if we're all being throttled? Is that possible?

**Comment by pdlove** [comment]
Try this with plugin version 0.4.4. I've corrected a lot on the error handling. You'll still get the connection errors because that is all smart things being an issue.... but it shouldn't cause home bridge to crash as it has been doing.

**Comment by pdlove** [comment]
I'm moving all discussion to issue #58. I was able to iron out one major issue last night an just released that as 0.4.6.
